pkg/motor/motor.go

```diff
@@ -30,46 +33,46 @@ type motorNodeImpl struct {
 	DIDDocument did.Document
 	SonrHost    host.SonrHost
 
+	// internal protocols
+	isHostEnabled      bool
+	isDiscoveryEnabled bool
+	callback           common.MotorCallback
+	discovery          *dp.DiscoverProtocol
+
+	// configuration
+	homeDir    string
+	supportDir string
+	tempDir    string
+
 	// Sharding
 	deviceShard   []byte
 	sharedShard   []byte
 	recoveryShard []byte
 	unusedShards  [][]byte
 
-	// query clients
-	schemaQueryClient st.QueryClient
-	bucketQueryClient bt.QueryClient
-
 	// resource management
 	Resources *motorResources
 }
 
-func EmptyMotor(id string) *motorNodeImpl {
+func EmptyMotor(r *mt.InitializeRequest, cb common.MotorCallback) *motorNodeImpl {
 	return &motorNodeImpl{
-		DeviceID: id,
+		isHostEnabled:      r.GetEnableHost(),
+		isDiscoveryEnabled: r.GetEnableDiscovery(),
+		DeviceID:           r.GetDeviceId(),
```

Comment:
We should add validation here to make sure the DeviceID was provided.

Reply:
yes ill make it GetDeviceID - this gives zero value and prevents motor node for panicing

Reply:
Wait it already does this - should EmptyMotor return an error if no deviceID provided?

Reply:
I think so, because there will be no way to uniquely identify the device with vault.

Reply:
if this is for our credentials and is expected to be user defined, throwing an error feels best. but if this is an internal id for p2p communication probably best left to motor to generate